jscomp/others/js_option.ml

```diff
@@ -59,5 +59,27 @@ let andThen f x =
   | None -> None 
   | Some x -> f x [@bs]
 
+let map f x =
+  match x with
+  | None -> None
+  | Some x -> Some (f x [@bs])
 
+let orValue a x =
+  match x with
+  | None -> a
+  | Some x -> x
 
+let filter f x =
+  match x with
+  | None -> None
```

Comment:
I use this all the time in Scala and Swift, to further narrow options. Usually filtering empty strings, for instance. I've seen it in every stdlib of languages that support options.